src/lib/fountain/test/fountain_sinkSpecialTest.cpp

```diff
@@ -0,0 +1,93 @@
+/* This code is subject to the terms of the Mozilla Public License, v.2.0. http://mozilla.org/MPL/2.0/. */
+#include "unittest.h"
+
+#include "FountainMetadata.h"
+#include "fountain_encoder_stream.h"
+#include "fountain_decoder_sink.h"
+
+#include "serialize/format.h"
+#include "util/File.h"
+#include "util/MakeTempDirectory.h"
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+using std::string;
+using namespace std;
+
+namespace {
+	stringstream dummyContents(unsigned size, string contents="0123456789")
+	{
+		stringstream input;
+		for (unsigned i = 0; i < (size/contents.size()); ++i)
+			input << contents;
+		return input;
+	}
+
+	string createFrame(fountain_encoder_stream& fes)
+	{
+		stringstream ss;
+		std::array<char, 125> buff;
+
+		// for a 7500 byte frame, 125*60 does the trick
+		for (int i = 0; i < 60; ++i)
+		{
+			unsigned res = fes.readsome(buff.data(), buff.size());
+			assertEquals( res, buff.size() );
+			ss << string(buff.data(), buff.size());
+		}
+		return ss.str();
+	}
+
+	std::string random_string(unsigned len, std::string chars="abcdefghijklmnaoqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890")
+	{
+		std::cout << ::rand() << std::endl;
+		std::string output = "";
+		for (unsigned i = 0; i < len; ++i)
+		{
+			unsigned idx = ::rand() % chars.length();
+			output += chars[idx];
+		}
+		return output;
+	}
+}
+
+TEST_CASE( "FountainSinkSpecialTest/testMultipart", "[unit]" )
+{
+	::srand( ::time(nullptr) );
+	MakeTempDirectory tempdir;
+
+	fountain_decoder_sink<std::ofstream> sink(tempdir.path(), 750);
+
+	const int totalSize = 6000000;
```

Comment:
The idea here is to permute through various random(ish) data permutations to see if anything trips up the decode. So far I've seen no issues.